sdk/protocol-plugins/src/plugins/common/helpers/aaveV3Like/AAVEv3LikeBaseProtocolPlugin.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  CollateralInfo,
+  DebtInfo,
+  Denomination,
+  FiatCurrency,
+  Maybe,
+  Percentage,
+  Price,
+  RiskRatio,
+  RiskRatioType,
+  TokenAmount,
+} from '@summerfi/sdk-common'
+import { BaseProtocolPlugin } from '../../../../implementation/BaseProtocolPlugin'
+import { AAVEv3LikeAbiInfo } from './AAVEv3LikeAbi'
+import {
+  AaveV3LikeProtocolDataBuilder,
+  filterAssetsListByEMode,
+} from './AAVEv3LikeProtocolDataBuilder'
+import { AllowedProtocolNames } from './AAVEv3LikeBuilderTypes'
+import { BigNumber } from 'bignumber.js'
+import { PRECISION_BI, UNCAPPED_SUPPLY } from '../../constants/AaveV3LikeConstants'
+import { CommonTokenSymbols, IToken } from '@summerfi/sdk-common/common'
+import { ICollateralInfo } from '@summerfi/sdk-common/protocols'
+
+type AssetsList = Awaited<ReturnType<AAVEv3LikeBaseProtocolPlugin['_getAssetsList']>>
+type Asset = AssetsList extends (infer U)[] ? U : never
+
+/**
+ * @class AAVEv3BaseProtocolPlugin
+ * @description Base class for AAVEv3 protocol plugins, it contains common functionality to
+ * fetch data from forks of AAVEv3 protocol.
+ */
+export abstract class AAVEv3LikeBaseProtocolPlugin extends BaseProtocolPlugin {
+  abstract readonly protocolName: AllowedProtocolNames
+  private _assetsList: Maybe<AssetsList> = undefined
+
+  /** PROTECTED */
+
+  /**
+   * Fetches the contract definition for the given contract name.
+   * @param contractName The name of the contract to fetch the definition for.
+   * @returns The contract definition for the given contract name.
+   *
+   * To be implemented by each specific protocol plugin.
+   */
+  protected abstract _getContractDef(contractName: string): AAVEv3LikeAbiInfo
+
+  /**
+   * Initializes the assets list if it hasn't been initialized yet.
+   *
+   * To be called before fetching data from the assets list, typically from `getLendingPool` and similar
+   * methods.
+   */
+  protected async _inititalizeAssetsListIfNeeded() {
+    if (this._assetsList) {
+      return
+    }
+
+    this._assetsList = await this._getAssetsList()
+  }
+
+  /**
+   * Fetches the assets list for the protocol.
+   * @returns The assets list for the protocol.
+   *
+   * This function must exist as the return type of the data builder is inferred from the return
+   * type of this function.
+   */
+  protected async _getAssetsList() {
+    try {
+      const _ctx = {
+        ...this.ctx,
+        getContractDef: this._getContractDef,
+      }
+      const builder = await new AaveV3LikeProtocolDataBuilder(_ctx, this.protocolName).init()
+      return await builder
+        .addPrices()
+        .addReservesCaps()
+        .addReservesConfigData()
+        .addReservesData()
+        .addEmodeCategories()
+        .build()
+    } catch (e) {
+      throw new Error(`Could not fetch/build assets list for AAVEv3: ${JSON.stringify(e)}`)
+    }
+  }
```

Comment:
We no longer have assets lists? What it does right now?

Reply:
It still fetches the assets from the protocol and then we filter it out to get the one that we need for the collateral and debt info. To be honest I didn't dare to refactor it to only get the required assets because of lack of time